### PR TITLE
[Bugfix] Support extending null builtin args

### PR DESCRIPTION
**Pull request by Tastyep**
# Description

Support [has been added](https://github.com/jose-elias-alvarez/null-ls.nvim/pull/106) to null-ls for extending builtins args :tada: 

## How Has This Been Tested?

``` lua
lvim.lang.sh.formatters = {
  {
    exe = "shfmt",
    args = { "-i", "2", "-ci" },
  },
}
```

### CODE REVIEW

**Comment by abzcoding** [comment]
do these args extend the default args in `null-ls` and if so, how can one go about replacing them instead of extending them?

**Comment by Tastyep** [comment]
> do these args extend the default args in `null-ls` and if so, how can one go about replacing them instead of extending them?

They do extend instead of overwrite, there is a discussion about it [here](https://github.com/jose-elias-alvarez/null-ls.nvim/issues/105)

TLDR: null-ls only provides arguments that are useful for itself [example](https://github.com/jose-elias-alvarez/null-ls.nvim/blob/main/lua/null-ls/builtins/diagnostics.lua#L251), these should be left unchanged.
I don;t see a situation where you would want to overwrite them, but you would do it like this: `builtin.with({args = your_args})`

**Review by abzcoding — APPROVED** [review]
seems to be working fine, but we should warn users about updating their null-ls or sth 😅

**Comment by Tastyep** [comment]
> seems to be working fine, but we should warn users about updating their null-ls or sth sweat_smile

Why ? I don't think it would break anything.

**Comment by abzcoding** [comment]
> > seems to be working fine, but we should warn users about updating their null-ls or sth sweat_smile
> 
> Why ? I don't think it would break anything.

some might have tried to override the `null-ls` config for example for `isort` instead of `--profile black` they had sth else, but now I don't know what happens since we are extending instead of overriding, but i think we should merge this anyways

**Comment by Tastyep** [comment]
This is an issue on null-ls side, I'll review the builtins tomorrow and PR removal of unnecessary parameters.
